lib/timers.js

```diff
@@ -501,7 +501,15 @@ function rearm(timer, start) {
     timer._handle.start(timer._repeat);
   } else {
     timer._idleTimeout = timer._repeat;
-    insert(timer, false, start);
+
+    const duration = TimerWrap.now() - start;
+    if (duration >= timer._repeat) {
+      // If callback duration >= timer._repeat,
+      // add 1 ms to avoid blocking eventloop
+      insert(timer, false, start + duration - timer._repeat + 1);
```

Comment:
I think just wrapping `insert()` in `process.nextTick()` should be enough. If not, perhaps just adding +1ms will work fine. (Timer granularity is only 1-2ms anyways even on fast machines afaik.)